solidity/contracts/bridge/library/Frauds.sol

```diff
@@ -0,0 +1,577 @@
+// SPDX-License-Identifier: MIT
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity ^0.8.9;
+
+import {BytesLib} from "@keep-network/bitcoin-spv-sol/contracts/BytesLib.sol";
+import {BTCUtils} from "@keep-network/bitcoin-spv-sol/contracts/BTCUtils.sol";
+import {CheckBitcoinSigs} from "@keep-network/bitcoin-spv-sol/contracts/CheckBitcoinSigs.sol";
+import "../EcdsaLib.sol";
+import "../Bridge.sol";
+
+library Frauds {
+    using BytesLib for bytes;
+    using BTCUtils for bytes;
+    using BTCUtils for uint32;
+    using EcdsaLib for bytes;
+
+    struct Data {
+        ///  The amount of stake slashed from each member of a wallet for a fraud.
+        uint256 slashingAmount;
+        /// The percentage of the notifier reward from the staking contract
+        /// the notifier of a fraud receives. The value is in the range [0, 100].
+        uint256 notifierRewardMultiplier;
+        /// The amount of time the wallet has to defeat a fraud challenge.
+        uint256 challengeDefeatTimeout;
+        /// The amount of ETH the party challenging the wallet for fraud needs
+        /// to deposit.
+        uint256 challengeDepositAmount;
+        /// Collection of all submitted fraud challenges indexed by challenge
+        /// key built as keccak256(walletPublicKey|sighash|v|r|s).
+        mapping(uint256 => FraudChallenge) challenges;
+    }
+
+    struct FraudChallenge {
+        // The address of the party challenging the wallet.
+        address challenger;
+        // The amount of ETH the challenger deposited.
+        uint256 depositAmount;
+        // The timestamp the challenge was submitted at.
+        uint32 reportedAt;
+        // The flag indicating whether the challenge has been closed.
+        bool closed;
+    }
+
+    /// @notice Represents Bitcoin signature in the R/S/V format.
+    struct RSVSignature {
```

Comment:
I wonder if it wouldn't make more sense to move this struct to `BitcoinTx` library. After all, it is nothing related to frauds, it's just a standard Bitcoin signature in R/S/V format.